File: 1.1/src/cz/mg/vulkan/PFNvkCmdBeginQuery.java
package cz.mg.vulkan;

public class PFNvkCmdBeginQuery extends VkFunctionPointer {
    public PFNvkCmdBeginQuery() {
    }

    protected PFNvkCmdBeginQuery(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdBeginQuery(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdBeginQuery(long value) {
        setValue(value);
    }

    public PFNvkCmdBeginQuery(VkInstance instance) {
        super(instance, new VkString("vkCmdBeginQuery"));
    }

    public void call(VkCommandBuffer commandBuffer, VkQueryPool queryPool, VkUInt32 query, VkQueryControlFlags flags){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), queryPool != null ? queryPool.getVkAddress() : VkPointer.getNullAddressNative(), query != null ? query.getVkAddress() : VkPointer.getNullAddressNative(), flags != null ? flags.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long queryPool, long query, long flags);


    public void call(VkCommandBuffer commandBuffer, VkQueryPool queryPool, int query, int flags){
        callSimplifiedNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), queryPool != null ? queryPool.getVkAddress() : VkPointer.getNullAddressNative(), query, flags);
    }

    protected static native void callSimplifiedNative(long vkaddress, long commandBuffer, long queryPool, int query, int flags);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdBeginQuery.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdBeginQuery_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong queryPool, jlong query, jlong flags){
    (void)env;
    (void)jc;
    PFN_vkCmdBeginQuery f = (PFN_vkCmdBeginQuery)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkQueryPool*)jniLongToPointer(queryPool)),
        *((uint32_t*)jniLongToPointer(query)),
        *((VkQueryControlFlags*)jniLongToPointer(flags))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdBeginQuery_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong queryPool, jint query, jint flags){
    (void)env;
    (void)jc;
    PFN_vkCmdBeginQuery f = (PFN_vkCmdBeginQuery)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkQueryPool*)jniLongToPointer(queryPool)),
        (uint32_t)query,
        (VkQueryControlFlags)flags
    );
}

